### PR TITLE
added `Cake\Cache\Cache::clearAll()` method

### PR DESCRIPTION
**Pull request by mirko-pagliai**

### CODE REVIEW

**Comment by lorenzo** [comment]
Can you add a test for this new code?

File: src/Cache/Cache.php [hunk 1]
```diff
@@ -445,6 +445,25 @@ public static function clear($check = false, $config = 'default')
...
+     * @param bool $check if true will check expiration, otherwise delete all
+     * @return bool True if the cache of all configurations was successfully cleared, false otherwise
+     */
+    public static function clearAll($check = false)
+    {
+        $success = true;
+        
+        foreach (self::configured() as $config) {
+            if (!self::clear($check, $config)) {
+                $success = false;
```
Comment (markstory) [inline src/Cache/Cache.php]: Should you stop clearing once an engine has failed?

**Comment by mirko-pagliai** [comment]
@lorenzo I'm not sure I know how to write a good test. I've read the documentation. Can I see some examples?

**Comment by mirko-pagliai** [comment]
@markstory I think is not necessary. But tell me ...

**Comment by markstory** [comment]
@mirko-pagliai I guess it depends on how you see this method being used. If a cache fails to clear should clearing continue, or should it stop and let the caller handle the failure. With the current code multiple cache backends could fail, and you have no idea which parts worked and which parts didn't.

**Comment by mirko-pagliai** [comment]
You're right. But this method calls the `clear()` method. So, the `clear()` method should throw an exception if it fails

**Comment by markstory** [comment]
@mirko-pagliai Returning an array of status codes could help mitigate the partial failure scenarios.

**Comment by mirko-pagliai** [comment]
@markstory What do you mean with "status codes", in this case?
This method calls `Cake\Cache\Cache::clear()` for each cache configuration. Each single call returns a boolean.

So, I can (some idea...):

1. return `TRUE` on success or an array with the names of the configurations that have failed on failure;
2. return an array with the names of the configurations as keys and a boolean (success or failure per that configuration) as value;
3. set some property like `$this->lastError`.

But I'm not convinced by any of these possibilities, I am sincere.

**Comment by markstory** [comment]
I meant an array of results like:

```php
[ 'slow' => true, 'fast' => false, 'blog' => true']
```

As the result of this method.

**Comment by mirko-pagliai** [comment]
Ok, so the second. I will update the PR ;)

**Comment by mirko-pagliai** [comment]
Done @markstory  Is it ok for you?

**Comment by markstory** [comment]
Looks good. New public methods/features need tests before they can be merged though.

**Comment by mirko-pagliai** [comment]
Yes, I know. I will try :-) be patient, I'm a noob!

**Comment by markstory** [comment]
@mirko-pagliai Not a problem. If you get stuck with the tests let me know.

**Comment by mirko-pagliai** [comment]
Hi @markstory, so I have to create a method like `testClearAll()` into  `tests/TestCase/Cache/CacheTest.php`

What should do this method?
If I understand correctly, something like:
1) add a new cache configuration;
2) write some data in that configuration;
3) delete the entire cache;
4) check whether the data has been deleted.

Did I get it right?

**Comment by markstory** [comment]
@mirko-pagliai Sounds good.

**Comment by mirko-pagliai** [comment]
I hope it goes well. This was the first test that I wrote :-)

**Comment by markstory** [comment]
Thanks :+1:

**Pull request merged**